### PR TITLE
Initial NUOPC cap

### PR DESCRIPTION
**Pull request by zmoon**
This initial effort replicates the functionality of the current standalone `nexus` executable, replacing it.

Differences:
* HEMCO time is controlled by an ESMF clock created in the NUOPC driver
* After NEXUS initialization, detected output fields are NUOPC-advertised and -realized
  - Currently, NEXUS initialization (setting up HEMCO, creating ESMF fields for the "Diag" and "Expt" outputs, etc.) is called in the app before the NUOPC driver starts, but this will probably need to change for real coupling situations
* Some small enhancements to the CLI and info messages

Most of the code was taken from the original `nexus_methods_mod` and `nexus_driver`, adapting where necessary. An attempt was made to:
* normalize the internal API (`nxs_*`, `hco_*`, ...)
* move to mostly procedure-scoped `use`-`only` imports from HEMCO (though still `use`ing ESMF and NUOPC at module level)
* `use` routines from `HCOI_StandAlone_Mod` where possible instead of duplicating them (but not all are public)
  - This is why there is a net line count decrease. Maybe we could get HEMCO to make more of those routines public to improve this more.
* normalize indentation (`findent -i2`)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
 if(NOT DEFINED ENV{CMAKE_Fortran_COMPILER})
   message(FATAL_ERROR "CMAKE_Fortran_COMPILER is not defined")
 endif()
 
 set(CMAKE_Fortran_COMPILER $ENV{CMAKE_Fortran_COMPILER})
 
-# set the project name and version
+# Set the project name and version
 project(NEXUS
         VERSION 0.1.0
```
Comment (zmoon) [inline CMakeLists.txt]: Do we want to change the version?

File: src/nuopc/cap.F90 [hunk 2]
```diff
@@ -0,0 +1,3131 @@
...
+
+  integer, parameter :: rootPet = 0
+
+  type(ESMF_StaggerLoc), parameter :: staggerList(2) = &
+    (/ ESMF_STAGGERLOC_CENTER, ESMF_STAGGERLOC_CORNER /)
+
+  private
+
+  public SetServices, &
+    init, finalize, &
```
Comment (zmoon) [inline src/nuopc/cap.F90]: Maybe these should be `nxs_init` and `nxs_finalize`

File: CMakeLists.txt (hunk 1 above)
Comment (bbakernoaa) [reply CMakeLists.txt]: Sure.

File: src/nuopc/cap.F90 (hunk 2 above)
Comment (bbakernoaa) [reply src/nuopc/cap.F90]: I like that.

File: CMakeLists.txt (hunk 1 above)
Comment (zmoon) [reply CMakeLists.txt]: But to what? 0.2.0 or 0.1.1?

Comment (bbakernoaa) [reply CMakeLists.txt]: I think 0.2

**Review by bbakernoaa — APPROVED** [review]

File: src/nuopc/cap.F90 (hunk 2 above)
Comment (zmoon) [reply src/nuopc/cap.F90]: Unresolving until I change them

Comment (zmoon) [reply src/nuopc/cap.F90]: Changed in 4a20c97

File: CMakeLists.txt (hunk 1 above)
Comment (zmoon) [reply CMakeLists.txt]: To 0.2 in 527c3d3

**Pull request merged**